tck/src/main/java/org/eclipse/microprofile/openapi/tck/ModelConstructionTest.java

```diff
@@ -1457,6 +1457,22 @@ public void parameterTest() {
         Example otherExampleValue = createConstructibleInstance(Example.class);
         checkMapImmutable(p, Parameter::getExamples, "otherExample", otherExampleValue);
         checkNullValueInAdd(p::getExamples, p::addExample, "otherExample", exampleValue);
+
+        // Test combination of Parameter.type = object and spaceDelimited/pipeDelimited
+        Schema schema = createConstructibleInstance(Schema.class);
+        schema.setType(Collections.singletonList(Schema.SchemaType.OBJECT));
+
+        final Parameter p2 = processConstructible(Parameter.class);
+        p2.setSchema(schema);
+        p2.setStyle(Parameter.Style.SPACEDELIMITED);
+        assertEquals(p2.getSchema().getType().get(0), Schema.SchemaType.OBJECT);
+        assertEquals(p2.getStyle(), Parameter.Style.SPACEDELIMITED);
+
+        final Parameter p3 = processConstructible(Parameter.class);
+        p3.setSchema(schema);
+        p3.setStyle(Parameter.Style.PIPEDELIMITED);
+        assertEquals(p3.getSchema().getType().get(0), Schema.SchemaType.OBJECT);
+        assertEquals(p3.getStyle(), Parameter.Style.PIPEDELIMITED);
```

Comment:
The `ModelConstructionTest` isn't the right place to test this. That would catch whether the setters immediately reject the value, but given that this restriction affects two parameters, it's much more likely that an implementation would validate it when the model is complete, rather than during construction.

Secondly, this only tests the model and doesn't test that this combination can be configured using annotations.

Reply:
I think to test this we really want a resource method with a parameter in this format.